### PR TITLE
Changes to support transpose intrinsic during initialization

### PR DESCRIPTION
**Pull request by kiranchandramohan**
Transpose intrinsic is a special case of the reshape intrinsic
with the order parameter set to /2,1/. We reuse the functions
for reshape intrinsic with minor additions wherever possible.

### CODE REVIEW

File: test/f90_correct/inc/transpose_init.mk [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#
+# Copyright (c) 2016, NVIDIA CORPORATION.  All rights reserved.
```
Comment (kiranchandramohan) [inline test/f90_correct/inc/transpose_init.mk:2]: Remove copyright and switch to new LLVM license. Here and in all new files.

File: tools/flang1/flang1exe/semutil2.c [hunk 2]
```diff
@@ -9857,11 +9880,16 @@ eval_reshape(ACL *arg, DTYPE dtype)
...
 
   if (orderarg == NULL) {
-    if (src_sz == dest_sz) {
-      return srclist;
-    }
-    for (i = 0; i < rank; i++) {
-      order[i] = i;
+    if (transpose) {
+        order[0] = 1;
+        order[1] = 0;
```
Comment (bryanpkc) [inline tools/flang1/flang1exe/semutil2.c:9885]: Does the semantic analysis for this initializer need to check that the ranks of the input and output are both 2? Or is that check already performed somewhere else?

Comment (kiranchandramohan) [reply tools/flang1/flang1exe/semutil2.c:10320]: Yes, sema does that check here for the input. https://github.com/flang-compiler/flang/blob/27fd9b2ec77be73c9bd90856056dab33af987c39/tools/flang1/flang1exe/semfunc.c#L7058
The output is not strictly checked probably even for non-constant transpose.

**Comment by pawosm-arm** [comment]
@bryanpkc can this one be merged?

File: tools/flang1/flang1exe/semutil2.c (hunk 2 above)
Comment (bryanpkc) [reply tools/flang1/flang1exe/semutil2.c:10320]: Thanks.

**Review by bryanpkc — APPROVED** [review]

**Pull request merged**